src/cachetools/__init__.py

```diff
@@ -497,78 +499,110 @@ def __getlink(self, key):
         return value
 
 
+def _get_function_value_from_cache(cache, key, lock=None):
+    lock = lock or contextlib.nullcontext()
+    try:
+        with lock:
+            return cache[key]
+    except KeyError:
+        return None
```

Comment:
If I'm right, this changes the behavior of the decorator to not  cache None values.